crates/ffmpeg/src/io/internal.rs

```diff
@@ -259,3 +259,145 @@ impl Inner<()> {
         Ok(this)
     }
 }
+
+#[cfg(test)]
+#[cfg_attr(all(test, coverage_nightly), coverage(off))]
+mod tests {
+    use std::fs;
+    use std::io::Cursor;
+    use std::path::Path;
+
+    use ffmpeg_sys_next::AVSEEK_FORCE;
+    use libc::{c_void, SEEK_CUR, SEEK_END};
+
+    use crate::error::FfmpegError;
+    use crate::io::internal::{read_packet, seek, write_packet, Inner, InnerOptions, AVERROR_EOF};
+
+    #[test]
+    fn test_read_packet_eof() {
+        let data: Cursor<Vec<u8>> = Cursor::new(vec![]);
+        let mut opaque = Box::new(data);
+        let mut buf = [0u8; 10];
+
+        unsafe {
+            let result =
+                read_packet::<Cursor<Vec<u8>>>(&mut *opaque as *mut _ as *mut c_void, buf.as_mut_ptr(), buf.len() as i32);
+
+            assert_eq!(result, AVERROR_EOF);
+        }
+    }
```

Comment:
This is unsafe & unsound I think.

I think all you need to do here is take the 
```
(&raw mut data) as *mut libc::c_void
```

You do not need to sue a box and deref the box.